### PR TITLE
Update benchmarks

### PR DESCRIPTION
**Pull request by adamdossa**
Updates benchmarks for 5.2

### CODE REVIEW

File: pallets/weights/src/pallet_staking.rs [hunk 1]
```diff
@@ -184,48 +187,49 @@ impl pallet_staking::WeightInfo for WeightInfo {
...
             .saturating_add(DbWeight::get().reads(3 as Weight))
             .saturating_add(DbWeight::get().writes(3 as Weight))
     }
     // Storage: Staking ValidatorCount (r:0 w:1)
-    fn set_validator_count(c: u32) -> Weight {
-        (3_184_000 as Weight)
-            // Standard Error: 0
-            .saturating_add((1_000 as Weight).saturating_mul(c as Weight))
+    fn set_validator_count(_c: u32, ) -> Weight {
+        (5_416_000 as Weight)
```
Comment (Neopallium) [inline pallets/weights/src/pallet_staking.rs]: This ignored parameter is fine.  Our benchmark shouldn't even have this parameter.

File: pallets/weights/src/pallet_staking.rs [hunk 2]
```diff
@@ -302,10 +306,8 @@ impl pallet_staking::WeightInfo for WeightInfo {
     // Storage: Staking Ledger (r:1 w:1)
     // Storage: Balances Locks (r:1 w:1)
     // Storage: System Account (r:1 w:1)
-    fn rebond(u: u32) -> Weight {
-        (78_101_000 as Weight)
-            // Standard Error: 2_000
-            .saturating_add((1_000 as Weight).saturating_mul(u as Weight))
+    fn rebond(_u: u32, ) -> Weight {
+        (76_450_000 as Weight)
```
Comment (Neopallium) [inline pallets/weights/src/pallet_staking.rs]: A fixed maximum (32) is used for the `u` parameter, so only `32_000` weight is removed by ignoring it.  I think this is fine.

File: pallets/weights/src/polymesh_contracts.rs [hunk 3]
```diff
@@ -242,12 +242,12 @@ impl polymesh_contracts::WeightInfo for WeightInfo {
...
     fn dummy_contract() -> Weight {
-        (468_460_000 as Weight)
+        (282_229_000 as Weight)
             .saturating_add(DbWeight::get().reads(12 as Weight))
             .saturating_add(DbWeight::get().writes(2 as Weight))
     }
-    fn basic_runtime_call(_n: u32) -> Weight {
-        (1_440_000 as Weight)
+    fn basic_runtime_call(_n: u32, ) -> Weight {
+        (1_570_000 as Weight)
```
Comment (Neopallium) [inline pallets/weights/src/polymesh_contracts.rs]: Ignoring the `n` parameter is fine.  Most likely we don't even need the `n` parameter in the benchmark.  This weight function is to get the overhead of making a cheap runtime call.

File: pallets/weights/src/pallet_identity.rs [hunk 4]
```diff
@@ -176,38 +177,36 @@ impl pallet_identity::WeightInfo for WeightInfo {
...
     }
     // Storage: Identity KeyRecords (r:2 w:1)
     fn set_secondary_key_permissions() -> Weight {
-        (53_854_000 as Weight)
+        (84_890_000 as Weight)
             .saturating_add(DbWeight::get().reads(2 as Weight))
             .saturating_add(DbWeight::get().writes(1 as Weight))
     }
-    fn permissions_cost(a: u32, p: u32, l: u32, e: u32) -> Weight {
+    fn permissions_cost(_a: u32, p: u32, l: u32, e: u32, ) -> Weight {
```
Comment (Neopallium) [inline pallets/weights/src/pallet_identity.rs]: These parameters are important and shouldn't be ignore.  The `a` parameter is for the number of asset permissions.

File: pallets/weights/src/pallet_asset.rs [hunk 5]
```diff
@@ -111,12 +111,12 @@ impl pallet_asset::WeightInfo for WeightInfo {
     // Storage: Asset Identifiers (r:0 w:1)
     // Storage: ExternalAgents AgentOf (r:0 w:1)
     // Storage: ExternalAgents GroupOfAgent (r:0 w:1)
-    fn create_asset(_n: u32, i: u32, f: u32) -> Weight {
-        (207_235_000 as Weight)
-            // Standard Error: 10_000
-            .saturating_add((66_000 as Weight).saturating_mul(i as Weight))
-            // Standard Error: 41_000
-            .saturating_add((100_000 as Weight).saturating_mul(f as Weight))
+    fn create_asset(_n: u32, i: u32, f: u32, ) -> Weight {
```
Comment (Neopallium) [inline pallets/weights/src/pallet_asset.rs]: The `n` parameter is the asset name length.  The tx per-byte fee should cover this.

File: pallets/weights/src/pallet_asset.rs [hunk 6]
```diff
@@ -299,18 +299,16 @@ impl pallet_asset::WeightInfo for WeightInfo {
...
     // Storage: Identity KeyRecords (r:1 w:0)
     // Storage: Asset CustomTypesInverse (r:1 w:1)
     // Storage: Asset CustomTypeIdSequence (r:1 w:1)
     // Storage: Asset CustomTypes (r:0 w:1)
-    fn register_custom_asset_type(n: u32) -> Weight {
-        (71_836_000 as Weight)
-            // Standard Error: 1_000
-            .saturating_add((6_000 as Weight).saturating_mul(n as Weight))
+    fn register_custom_asset_type(_n: u32, ) -> Weight {
+        (61_694_000 as Weight)
```
Comment (Neopallium) [inline pallets/weights/src/pallet_asset.rs]: `n` is the length of the type name.  Ignoring it is not critical.

File: pallets/weights/src/pallet_sto.rs [hunk 7]
```diff
@@ -60,15 +60,14 @@ impl pallet_sto::WeightInfo for WeightInfo {
...
     // Storage: Portfolio PortfolioAssetBalances (r:1 w:0)
     // Storage: Portfolio PortfolioLockedAssets (r:1 w:1)
     // Storage: Sto FundraiserNames (r:0 w:1)
     // Storage: Sto Fundraisers (r:0 w:1)
-    fn create_fundraiser(i: u32) -> Weight {
-        (122_668_000 as Weight)
-            // Standard Error: 1_610_000
-            .saturating_add((4_429_000 as Weight).saturating_mul(i as Weight))
-            .saturating_add(DbWeight::get().reads(10 as Weight))
+    fn create_fundraiser(_i: u32, ) -> Weight {
```
Comment (Neopallium) [inline pallets/weights/src/pallet_sto.rs]: The `i` parameter is the number of price tiers.  This shouldn't be ignored.

File: pallets/weights/src/pallet_checkpoint.rs [hunk 8]
```diff
@@ -86,10 +87,8 @@ impl pallet_asset::checkpoint::WeightInfo for WeightInfo {
     // Storage: Checkpoint TotalSupply (r:0 w:1)
     // Storage: Checkpoint Timestamps (r:0 w:1)
     // Storage: Checkpoint ScheduleRefCount (r:0 w:1)
-    fn create_schedule(s: u32) -> Weight {
-        (137_723_000 as Weight)
-            // Standard Error: 106_000
-            .saturating_add((518_000 as Weight).saturating_mul(s as Weight))
+    fn create_schedule(_s: u32, ) -> Weight {
+        (147_224_000 as Weight)
```
Comment (Neopallium) [inline pallets/weights/src/pallet_checkpoint.rs]: A fixed value of `1` is passed to this weight function.  So ignoring it here and the new weights is fine.  But we should review the benchmarks for this.

**Review by Neopallium — CHANGES_REQUESTED** [review]

File: pallets/weights/src/pallet_identity.rs (hunk 4 above)
Comment (adamdossa) [reply pallets/weights/src/pallet_identity.rs]: Copied over:
```
            // Standard Error: 60_000
            .saturating_add((296_000 as Weight).saturating_mul(a as Weight))        
```

File: pallets/weights/src/pallet_sto.rs (hunk 7 above)
Comment (adamdossa) [reply pallets/weights/src/pallet_sto.rs]: Copied over:
`            .saturating_add((4_429_000 as Weight).saturating_mul(i as Weight))`

**Comment by adamdossa** [comment]
@Neopallium branch has been updated via `cargo fmt` and in addition the benchmarks for:
```
create_fundraiser
permissions_cost
```
have been updated to include weights for the previously unignored parameters.

File: pallets/weights/src/pallet_sto.rs (hunk 7 above)
Comment (Neopallium) [reply pallets/weights/src/pallet_sto.rs]: I am not seeing this change.

File: pallets/weights/src/pallet_identity.rs (hunk 4 above)
Comment (Neopallium) [reply pallets/weights/src/pallet_identity.rs]: This is still missing.

**Review by Neopallium — APPROVED** [review]

**Pull request merged**